taichi/program/sparse_matrix.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include "taichi/common/core.h"
+#include "taichi/inc/constants.h"
+#include "Eigen/Sparse"
+
+namespace taichi {
+namespace lang {
+
+class SparseMatrix;
+
+class SparseMatrixBuilder {
+ public:
+  SparseMatrixBuilder(int n, int m, int max_num_triplets);
+
+  void *get_data_base_ptr();
+
+  void print_triplets();
+
+  void build(SparseMatrix *sm);
```

Comment:
Once `std::unique_ptr` is sorted out, let's just return a `std::unique_ptr<SparseMatrix>`?

Reply:
It's a good idea. I just return a `SparseMatrix` object.